streamer/parser/metadata.md

```diff
@@ -111,52 +105,59 @@ The contributor object may also contain a `sortAs` string, used to sort the cont
 When parsing an EPUB, we need to establish:
 
 * the key of the contributor;
-* the name of this contributor;
-* the alternate forms for this name;
+* a language map for the name of this contributor;
 * the string used to sort the name of the contributor.
 
 ### EPUB 2.x
 
 The following mapping should be used to determine the key of the contributor’s object: 
 
-| element        | opf:role               | key         |
-|----------------|------------------------|-------------|
-| dc:creator     | aut                    | author      |
-| dc:contributor | trl                    | translator  |
-| dc:contributor | est                    | editor      |
-| dc:contributor | ill                    | illustrator |
-| dc:contributor | art                    | artist      |
-| dc:contributor | clr                    | colorist    |
-| dc:contributor | nrt                    | narrator    |
-| dc:contributor | \<empty\> or \<other\> | contributor |
+| element                      | opf:role                 | key         |
+|------------------------------|--------------------------|-------------|
+| dc:creator                   | \<empty\> or \<unknown\> | author      |
+| dc:creator or dc:contributor | aut                      | author      |
+| dc:contributor               | \<empty\> or \<unknown\> | contributor |
+| dc:publisher                 | \<any\>                  | publisher   |
+| dc:creator or dc:contributor | pbl                      | publisher   |
+| dc:creator or dc:contributor | trl                      | translator  |
+| dc:creator or dc:contributor | edt                      | editor      |
+| dc:creator or dc:contributor | ill                      | illustrator |
+| dc:creator or dc:contributor | art                      | artist      |
+| dc:creator or dc:contributor | clr                      | colorist    |
+| dc:creator or dc:contributor | nrt                      | narrator    |
 
 Where `opf:role` is the value of the attribute of the `<dc:element>`.
 
-The `name` of the contributor is the value of the element.
+In case of conflict, `opf:role` overrides the XML element used. For example, map a `<dc:publisher>` or `<dc:contributor>` element whose value for `opf:role` is `aut` to an author.
```

Comment:
```suggestion
As a rule of thumb, `opf:role` overrides `<dc:creator>` and `<dc:contributor>`. For example, map a `<dc:contributor>` element whose value for `opf:role` is `aut` to an author.
```

Reply:
I'm not sure about that. <dc:creator> doesn't imply a key by itself, and we decided that there is no clear distinction between `dc:creator` and `dc:contributor` too. So what is overridden?

Reply:
We could use something else than `override`. I’m not sure how to rephrase this because I honestly can’t even tell whether I’ve been using those right or wrong for all those years at this point.

And, should we change this mapping then?

| element | opf:role | key |
|---------|----------|-----|
| dc:creator | \<empty\> or \<unknown\> | author |

It was my understanding this is how we decided to handle:

> It's not clear from the table how these elements are mapped when the opf:role attribute is missing. Is <dc:creator> ending up in authors and <dc:contributor> in contributors, like before?

But maybe it should be explored a little bit more – if it’s `<unknown>` should it be a `contributor` object with a `role`?

I mean I’ve just checked and a `<dc:creator>` without an `aut` refine passes EPUBCheck so we must have a default mapping for `<empty>`. But `<unknown>` I’m not so sure now.

Reply:
> In case of conflict, `opf:role` overrides the XML element used. For example, map a `<dc:publisher>` or `<dc:contributor>` element whose value for `opf:role` is `aut` to an author.

I think this sentence is simply useless with the updated table. All cases are covered.

It's hard to figure out a `<dc:creator>` with an unknown role. Maybe `abridger` or `interviewee` are enough important to be used with a `<dc:creator>` element, but should not be mapped to `author`.

Reply:
Sounds reasonable, let’s remove those 2 paragraphs then. 

For documentation’s sake, if that can be useful in the future, a few references to [the EPUB 3.2 spec](https://www.w3.org/publishing/epub3/epub-packages.html#sec-opf-dccreator) (emphasis mine).

> The [DC11] creator element represents the name of a person, organization, etc. responsible for the creation of the content of the Rendition. The `role` property **can** be attached to the element to indicate the function the creator played in the creation of the content.

> The creator element SHOULD contain the name of the creator as the Author **intends it to be displayed to a user.**

There’s so much room for interpretation that I won’t firmly state what an `<empty>` role means but well at least it seems reasonable to me that in most cases, it will be the name on the cover and in the page header of a print book.

And thanks to everyone for their patience.